tracer/src/Datadog.Trace/ClrProfiler/CallTarget/CallTargetState.cs

```diff
@@ -21,6 +22,8 @@ namespace Datadog.Trace.ClrProfiler.CallTarget
         private readonly object _state;
         private readonly DateTimeOffset? _startTime;
 
+        private readonly IReadOnlyDictionary<string, string> _previousDistributedSpanContext;
```

Comment:
This struct is getting big, maybe we should consider passing by red to the EndMethod.

Reply:
@tonyredondo would you like me to pass this by ref before merging? It seems like a good change to make in this PR

Reply:
We can do that, but remember we need to do this without removing the current support (because is a breaking change), something like we did for calltarget byref. By enabling the ByRef support with the PInvoke call. We can use the same pinvoke to switch to this optimization.

Reply:
If you want I can do that modification later in a separated PR.

Reply:
Ok let's do that, separate PR 👍🏼

Reply:
Just catching up, so for prosperity, this was done in #2169